src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -60,6 +60,14 @@ using std::auto_ptr;
 
 /* ****************************************************************************
 *
+* OtisTreatFunction - callback signature for treatOntimeintervalSubscriptions
+*/
+typedef void (*OtisTreatFunction)(std::string tenant, BSONObj* bobjP);
```

Comment:
The crucial point is, is the object modified or not?
This is a callback, and a 'callbacker' shouldn't know whether the object will be modified or not.

However, this is entirely up to us.
I find it confusing (just as they explain) to send what seems to be a copy of an object on the stack, and later find that the 'original' object has been changed.
I can see that you want to use this technique, so I'll just change it to a C++ reference.
No reason to waste any more time on this.